src/averageAges.js

```diff
@@ -59,6 +81,29 @@ function calculateAverageAgeDiff(people, onlyWithSon) {
   // 2. keep people who have mothers in the array
   // 3. calculate the difference child.born - mother.born
   // 4. return the average value
+  let haveMothers = [];
+
+  if (onlyWithSon) {
+    haveMothers = people.filter(person => person.mother
+    && person.sex === 'm');
+  } else {
+    haveMothers = people.filter(person => person.mother);
+  }
+
+  const ageDifferences = [];
+
+  for (const person of haveMothers) {
```

Comment:
Rewrite this part of code using array methods 
![image](https://github.com/mate-academy/js_average-ages/assets/70009205/924a9a68-1380-4101-9db8-cdcbdb401179)